packages/eslint-plugin/docs/rules/no-unused-expressions.mdx

```diff
@@ -10,4 +10,8 @@ import TabItem from '@theme/TabItem';
 > See **https://typescript-eslint.io/rules/no-unused-expressions** for documentation.
 
 This rule extends the base [`eslint/no-unused-expressions`](https://eslint.org/docs/rules/no-unused-expressions) rule.
-It adds support for optional call expressions `x?.()`, and directive in module declarations.
+It supports TypeScript-specific expressions:
+
+- Handles instantiation expressions: `Set<number>;`
+- Handles assertion expressions: `x as number;`, `x!;`, `<number>x;`
```

Comment:
nit - rather than just "handles" maybe clarify whether assertion expressions are used or unused? (like in the subsequent bullet)

Reply:
It's unused if the underlying expression (i.e. the LHS) is unused, and not unused if the LHS is considered not (such as call expression)

Reply:
Oh, yeah, so it's basically just transparent to type-y syntax. 🤷 Hard to know how to word it. Probably fine as is, then!